backends/platform/atari/readme.txt

```diff
@@ -0,0 +1,422 @@
+  and makes a *huge* difference for 640x480 fullscreen updates.
+
+
+Performance considerations/pitfalls
+-----------------------------------
+
+It's important to understand what affects performance on our limited platform
+to avoid unpleasant playing experiences.
+
+Game engines with unexpected performance hit
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A typical example from this category is the Gobliins engine (and its
+sequels). At first it looks like our machine / backend is doing something
+terribly wrong but the truth is it is the engine itself which is doing a lot of
+unnecessary redraws and updates, sometimes even before reaching the backend.
+The only real solution is to profile and fix the engine.
+
+Too many fullscreen updates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Somewhat related to the previous point - sometimes the engine authors didn't
+realize the impact of every update on the overall performance and instead of
+updating only the rectangles that really had changed, they ask for a full screen
+update. Not a problem on a >1 GHz machine but very visible on Atari! Also, this
+is (by definition) the case of animated intros, especially those in 640x480.
+
+MIDI vs. AdLib vs. sampled music
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It could seem that sample music replay must be the most demanding one but on the
+contrary! _Always_ choose a CD version of a game (with *.wav tracks) to any
+other version. With one exception: if you have a native MIDI device able to
+replay the given game's MIDI notes (using the STMIDI plugin). MIDI emulation
+(synthesis) can easily take down as many as 10 FPS.
+
+CD music slows everything down
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some games use separate audio *and* video streams (files). Even if the CPU is
+able to handle both, the bottleneck becomes ... disk access. This is visible in
+The Curse Of Monkey Island for example -- there's audible stuttering during the
+intro sequence (and during the game as well). Increasing "output_samples" makes
+the rendering literally crawling! Why? Because disk I/O is busy with loading
+even *more* sample data so there's less time for video loading and rendering.
+Try to put "musdisk1.bun" and "musdisk2.bun" into a ramdisk (i.e. u:/ram in
+FreeMiNT), you'll be pleasantly surprised with the performance boost gained.
+
+"Mute" vs. "Mute all" in GUI vs. "No music" in GUI
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Not the same thing. "Mute" (available only via the shortcut CONTROL+u) generates
+an event to which the sample mixer can react (i.e. stop mixing the silence...).
+
+"Mute all" doesn't generate anything, it basically just lowers the volume of the
+music to zero.
+
+"No music" means using the null audio plugin which prevents generating any MIDI
+music (and therefore avoiding the expensive synthesis emulation) but beware, it
+doesn't affect CD (*.wav) playback at all!
+
+So for the best performance, always choose "No music" in the GUI options when
```

Comment:
As I mentioned, this must be thoroughly tested as we assume that there are regular update calls. We are freeing sound handlers during those.